geomesa-core/src/main/scala/org/locationtech/geomesa/core/iterators/TemporalDensityIterator.scala

```diff
@@ -30,125 +28,62 @@ import org.apache.commons.codec.binary.Base64
+                                                         env: IteratorEnvironment): Unit = {
 
     dateTimeFieldName = getDtgFieldName(simpleFeatureType).getOrElse ( throw new IllegalArgumentException("dtg field required"))
 
-    // default to text if not found for backwards compatibility
-    val encodingOpt = Option(options.get(FEATURE_ENCODING)).getOrElse(FeatureEncoding.TEXT.toString)
-    originalDecoder = SimpleFeatureDecoder(simpleFeatureType, encodingOpt)
-
-    projectedSFT = SimpleFeatureTypes.createType(simpleFeatureType.getTypeName, TEMPORAL_DENSITY_FEATURE_STRING)
-
-    temporalDensityFeatureEncoder = SimpleFeatureEncoder(projectedSFT, encodingOpt)
-    featureBuilder = ScalaSimpleFeatureFactory.featureBuilder(projectedSFT)
-
     val buckets = TemporalDensityIterator.getBuckets(options)
     val bounds = TemporalDensityIterator.getTimeBounds(options)
     snap = new TimeSnap(bounds, buckets)
-
   }
 
-  /**
-   * Combines the results from the underlying iterator stack
-   * into a single feature
-   */
-  def findTop() = {
-    // reset our 'top' (current) variables
-    result.clear()
-    topSourceKey = null
-    topSourceValue = null
-
-    while(source.hasTop && !curRange.afterEndKey(source.getTopKey)) {
-      topSourceKey = source.getTopKey
-      topSourceValue = source.getTopValue //SimpleFeature
-
-      val date = originalDecoder.decode(topSourceValue.get()).getAttribute(dateTimeFieldName).asInstanceOf[Date]
-      val dateTime = new DateTime(date.getTime)
-      addResultDate(dateTime)
-
-      source.next()
-    }
-
-    if(topSourceKey != null) {
-      featureBuilder.reset()
-      featureBuilder.add(TemporalDensityIterator.encodeTimeSeries(result))
-      featureBuilder.add(TemporalDensityIterator.zeroPoint) //Filler value as Feature requires a geometry
-      val feature = featureBuilder.buildFeature(Random.nextString(6))
-      topTemporalDensityKey = Some(topSourceKey)
-      topTemporalDensityValue = Some(new Value(temporalDensityFeatureEncoder.encode(feature)))
-    }
+  override def handleKeyValue(resultO: Option[TemporalDensityIteratorResult],
+                              topSourceKey: Key,
+                              topSourceValue: Value): TemporalDensityIteratorResult = {
+    val date = originalDecoder.decode(topSourceValue.get()).getAttribute(dateTimeFieldName).asInstanceOf[Date]
+    val dateTime = new DateTime(date.getTime)
+    val result = resultO.getOrElse(TemporalDensityIteratorResult())
+    addResultDate(dateTime, result.timeSeries)
+    result
   }
 
   /** take a given Coordinate and add 1 to the result time that it corresponds to via the snap time */
-  def addResultDate(date: DateTime) = {
+  def addResultDate(date: DateTime, result: TimeSeries): Unit = {
     val t: DateTime = snap.t(snap.i(date))
-    val cur: Long = result.get(t).getOrElse(0L)
+    val cur: Long = result.getOrElse(t, 0L)
     result.put(t, cur + 1L)
   }
 
   override def seek(range: ARange,
```

Comment:
shouldn't need this method anymore as it's implemented in the super class, right?